python/ray/autoscaler/_private/aws/node_provider.py

```diff
@@ -176,6 +178,40 @@ def external_ip(self, node_id):
 
         return node.public_ip_address
 
+    def get_node_id(self, ip_address, use_internal=False):
```

Comment:
Having to keep these IP maps is a much bigger change than I expected. Is there a way to keep this change small by having a common implementation only, or only implementing it for kubernetes?

Reply:
If this is identical for every provider, why not just implement it in the abstract node provider? 

Regardless, +1 to a common implementation (maybe explicitly define the cache abstraction?)